src/meta/protos/proto/connection.proto

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package databend_proto;
+
+message UserDefinedConnection {
+  uint64 ver = 100;
+  uint64 min_reader_ver = 101;
+
+  string name = 1;
+  string storage_type = 2;
+  map<string, string> storage_params= 3;
```

Comment:
Is map in proto3 ordered?

Reply:
you are right, thanks!

Reply:
> Is map in proto3 ordered?

not ordered. but we do not rely on this order.